### PR TITLE
New OSI input routine based on disassembly of ROM code updated according to review

### PR DESCRIPTION
**Pull request by smuehlst**
This pull request is the successor to the cancelled pull request #567 by @jefftranter. I rebased Jeff's commit to the current master branch and addressed the review comments made by @greg-king5 here:

https://github.com/cc65/cc65/pull/567#issuecomment-358406672

Jeff and I successfully tested the new implementation.

### CODE REVIEW

**Comment by oliverschmidt** [comment]
@greg-king5: As far as I understand you looked into this topic before. May I ask you to review this PR?

File: libsrc/osic1p/cgetc.s [hunk 1]
```diff
@@ -2,49 +2,40 @@
...
+        ldx     CHARBUF         ; character in buffer available?
+        bne     done
         lda     cursor          ; show cursor?
         beq     nocursor
         ldy     CURS_X
         lda     (SCREEN_PTR),y  ; fetch current character
         sta     tmp1            ; save it
         lda     #$A1            ; full white square
-        sta     (SCREEN_PTR),y  ; store at cursor position
+        sta     (SCREEN_PTR),y  ; store at cursor position^
```
[the remark this answers is not in the document]
Comment (smuehlst) [reply libsrc/osic1p/cgetc.s]: No, that was a typo, removed `^`.

File: libsrc/osic1p/kbhit.s [hunk 2]
```diff
@@ -10,38 +10,197 @@
...
+        and     #$01
+        tax
+        lda     KBDTMP
+        and     #$06
+        bne     LFDA2
+        bit     LASTSCAN
+        bvc     LFDBB
+        txa
+        eor     #$01
+        and     #$01
```
[the remark this answers is not in the document]
Comment (smuehlst) [reply libsrc/osic1p/kbhit.s]: I removed the redundant `AND` in line 109, and this way also the subsequent `LDA` became redundant and was removed.

**Comment by Fabrizio-Caruso** [comment]
Sorry if I ask. What is blocking the merge? I am just trying to understand.

**Comment by greg-king5** [comment]
> Sorry if I ask. What is blocking the merge? I am just trying to understand.

Absent-mindedness.  (I'll look at it right now.)

**Comment by oliverschmidt** [comment]
Thanks!

**Comment by smuehlst** [comment]
@greg-king5 Thanks from me as well!

**Comment by Fabrizio-Caruso** [comment]
Tested perfectly working with my universal game Cross Chase.
Thanks everyone! This make it possible to run my universal game in non-turn-based mode also on the OSIC1P targets!

**Pull request merged**